pkg/distributor/push.go

```diff
@@ -307,43 +307,67 @@ func calculateRetryAfter(retryAttemptHeader string, baseSeconds int, maxBackoffE
 // to that error, if the error is one of the errors from this package. Otherwise, codes.Internal and
 // http.StatusInternalServerError is returned.
 func toGRPCHTTPStatus(ctx context.Context, pushErr error, limits *validation.Overrides) (codes.Code, int) {
```

Comment:
Since we now have `toGRPCHTTPStatusOtlp()`, I would revert `toGRPCHTTPStatus()` to its version before [this change](https://github.com/grafana/mimir/pull/8227/files#diff-8a770bd239c72b7de7bc51e232145f906f94d15727124a15fbe0d46c2d98def7L201), since it was introduced only for otlp purposes.

Reply:
Another recommendation: there is already `distributor.toGRPCError()` function that converts distributor errors to gRPC errors. Can't we adapt that function to the needs of otlp? That function is not used (except maybe by the aggregator in backend-enterprise), so we can change the returned codes to adapt them to the new needs.

Reply:
It was a comment from oleg, the logic of checking context.Cancel is the same as default return. already revert it before the comment to have minimum diff.

Reply:
we just want to return a grpc status.Status object in the response body with the origin error, that was done in the same as above comments comes. The grpc error code is not even decoded in otlp collector, I don't think it worth the entire wrap logics and modify toGRPCError() instead. I would propose to focus on the bug fix in the PR.